src/MultivariateStats.jl

```diff
@@ -337,12 +339,116 @@ function MLJBase.predict(m::LDA, (core_res, class_list), Xnew)
+    nclasses   = length(class_list)
+
+    # NOTE: copy/transpose
+    Xm_t   = MLJBase.matrix(X, transpose=true) # now p x n matrix
+    yplain = MLJBase.int(y) # vector of n ints in {1,..., nclasses}
+    p, n   = size(Xm_t)
+   
+    # check output dimension default is min(p, nc-1)
+    def_outdim = min(p, nclasses - 1)
+    # if unset (0) use the default; otherwise try to use the provided one
+    out_dim = ifelse(model.out_dim == 0, def_outdim, model.out_dim)
+    # check if the given one is sensible
+    out_dim ≤ def_outdim || throw(ArgumentError("`out_dim` must not be larger than `min(p, nc-1)` where `p` is the dimension of `X` and `nc` is the number of classes."))
+
+    ## Estimates prior probabilities is unspecified by user.
+    if model.priors == nothing
+        priors = proportions(yplain)
+    else
+     #check if the length of priors is same as nclasses
+     length(MLJBase.classes(model.priors)) == nclasses || throw(ArgumentError("Invalid size of `priors`"))
+     priors = MLJBase.pdf.(model.priors, class_list)
+    end
+
+    core_res = MS.fit(MS.MulticlassLDA, nclasses, Xm_t, Int.(yplain);
+                      method=model.method,
+                      outdim=out_dim,
+                      regcoef=model.regcoef,
+                      covestimator_within=model.cov_w,
+                      covestimator_between=model.cov_b)
+    
+    ## The original projection matrix satisfies Pᵀ*Sw*P=I
+    ## scaled projection_matrix and core_res.proj by multiplying by sqrt(n - nclasses) this ensures Pᵀ*Σ*P=I
+    ## where covariance estimate Σ = Sw / (n - nclasses)
+    core_res.proj   = core_res.proj .* sqrt(n - nclasses) 
+    core_res.pmeans = core_res.pmeans .* sqrt(n - nclasses)
+
+    cache     = nothing
+    report    = NamedTuple{}()
+    fitresult = (core_res, class_list, priors)
+
+    return fitresult, cache, report
+end
+
+function MLJBase.fitted_params(::BayesianLDA, (core_res, class_list, priors))
+    return (class_means       = MS.classmeans(core_res),
+            projection_matrix = MS.projection(core_res),
+            priors            = priors)
+end
+
+function MLJBase.predict(m::BayesianLDA, (core_res, class_list, priors), Xnew)
+    # projection of Xnew XWt is n x o  where o = number of out dims
+    XWt = MLJBase.matrix(Xnew) * core_res.proj
+    # centroids in the transformed space, nc x o
+    centroids = permutedims(core_res.pmeans)
+
+    n  = core_res.stats.tweight # n is the Number of training examples
+    nclasses = length(class_list) 
+
+    # compute the distances in the transformed space between pairs of rows
+    # The discriminant matrix `P` is of dimension `n x nc`
+    #  P[i,k] = -0.5*(xᵢ −  µₖ)ᵀΣ⁻¹(xᵢ −  µₖ) + log(priorsₖ) and Σ⁻¹ = I due to the nature of the projection_matrix
+
+    P = pairwise(SqEuclidean(), XWt, centroids, dims=1)
+    P .*= -0.5
+    P .+= log.(priors)'
+   
+    # apply a softmax transformation to convert P to a probability matrix
+    P  .= exp.(P)
```

Comment:
you need to use the maximum  trick (see in the other LDA) otherwise it may underflow /overflow

Reply:
I don't think there is need for this as the bayes formula prevents this. Besides this affects the predicted posterior probability distribution.